packages/server/src/automations/automationUtils.ts

```diff
@@ -96,6 +98,93 @@ export function getError(err: any) {
   return typeof err !== "string" ? err.toString() : err
 }
 
+export async function sendAutomationAttachmentsToStorage(
+  tableId: string,
+  row: Row
+) {
+  let table = await sdk.tables.getTable(tableId)
+  const attachmentRows: Record<string, AutomationAttachment[]> = {}
+
+  for (const [prop, value] of Object.entries(row)) {
+    const schema = table.schema[prop]
+    if (
+      schema?.type === FieldType.ATTACHMENTS ||
+      schema?.type === FieldType.ATTACHMENT_SINGLE
+    ) {
+      attachmentRows[prop] = value
+    }
+  }
+
+  for (const [prop, attachments] of Object.entries(attachmentRows)) {
+    if (attachments.length) {
+      row[prop] = await Promise.all(
+        attachments.map(attachment => generateAttachmentRow(attachment))
+      )
+    }
+  }
```

Comment:
Doesn't this omit generating single attachment rows? If the schema above is `FieldType.ATTACHMNENT_SINGLE` won't `value` be a non-array, so this `attachements.length` check will fail?

Reply:
You were right, I had changed this recently and forgot to add in the other case. Sorted.